ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/health/DropRepeated.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.health
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+
+class DropRepeated[T] extends GraphStage[FlowShape[T, T]] {
+  private val in = Inlet[T]("input")
+  private val out = Outlet[T]("DropRepeated output")
+
+  override def shape: FlowShape[T, T] = FlowShape(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) {
+      private var currentValue: Option[T] = None
```

Comment:
Pretty sure you don't need to; Akka manages synchronizations because actors are blocking. If you like we can loop in an expert (if you know of one hanging around).

Reply:
Cool, thanks!

Reply:
Should be fine inside `GraphStageLogic`.